### ISSUE TITLE
Agent Commission should be paid based on effective payment of Invoice

### ISSUE THREAD

**Issue by michelebr76** [asandulsky/1C-Italia#40]
We already opened an issue about the "Agent Commission" that should be calculated on value without VAT (and not on total).
IMPORTANT: it should be calculated on base of the effective payment of the invoice (it cannot be paid before).
It means that the Report Agent should have the following lines:
Total Document, Total Paid (of the document), Total Agent, Total Agent Effective (it is based on how much in paid). Total Paid to agent, Total to be paid.

**Comment by michelebr76** [comment]
This report should be ready at the END OF DECEMBER

**Comment by ValentinaKopeva** [comment]
Check it

**Comment by michelebr76** [comment]
This activities has been done but it is missing the report it was asked (i copy and paste what i wrote):

_It means that the Report Agent should have the following lines:
Total Document, Total Paid (of the document), Total Agent, Total Agent Effective (it is based on how much in paid). Total Paid to agent, Total to be paid._

it is possible to have it? Maybe I am just not able to do it myself

**Comment by michelebr76** [comment]
Today with Alexander we discussed about the Agent.
I send the actual report of Gruppo Simon regarding it

[Prova provvigioni luigi 2019.xlsx](https://github.com/asandulsky/1C-Italia/files/2765529/Prova.provvigioni.luigi.2019.xlsx)

**Comment by michelebr76** [comment]
I check and it seems nothing working about Agent.
I created a document:
![image](https://user-images.githubusercontent.com/37332394/51271116-f835b880-19c6-11e9-959a-23492a4c393f.png)



Contract has specified agent commission of 7%, i also valorized the name of agent and select the contract. I also paid money... but nothing in the report.
![image](https://user-images.githubusercontent.com/37332394/51271035-c886b080-19c6-11e9-82c0-56884a036c91.png)

**Comment by michelebr76** [comment]
I noticed that it has been inserted some registration with Agent (see here)

![image](https://user-images.githubusercontent.com/37332394/51353983-7ae77200-1ab2-11e9-8a07-4dec36c32dd8.png)

I tried to do myself, but it was not working. For sure it is my mistake.
In addition, not clear the botton "Generate Agent", when i press it nothing happened, and when I pressed on documents with agent value, it was put to zero.
Actually it is not needed any GenerateAgent button. Simply when there is an agent contract, it should be MANDATORY to specify the agent and calculate the agency %
